### PR TITLE
capture: store _suspended state

### PR DESCRIPTION
**Pull request by blueyed**
TODO:

- [x] squash

### CODE REVIEW

**Comment by blueyed** [comment]
This is useful for debugging, and is a starting point for making assertions (no nested suspend) in the future.

File: src/_pytest/capture.py [hunk 1]
```diff
@@ -495,6 +504,7 @@ def suspend_capturing(self, in_=False):
             self._in_suspended = True
 
     def resume_capturing(self):
+        self._suspended = False
```
Comment (nicoddemus) [inline src/_pytest/capture.py]: I noticed `stop_capturing` doesn't set `self._suspended = True` and I can see why, it is not really suspended after all, it has stopped. But that might mean we don't want a boolean, but a "state" string with `"capturing"`, `"suspended"` and `"stopped"`. What do you think?

Comment (blueyed) [reply src/_pytest/capture.py]: Makes sense.  Could also be set to `None` there though.

Comment (nicoddemus) [reply src/_pytest/capture.py]: Yep, but in that case `"stopped"` might be better for debugging purposes, and we can even start it with a proper string `"not initialized"` or something.

Comment (blueyed) [reply src/_pytest/capture.py]: Yeah.. note however that MultiCapture has `_reset` also already.
I suggest adding `_state` (string), and removing `_reset` then.

Comment (nicoddemus) [reply src/_pytest/capture.py]: Sounds good. 👍

Comment (blueyed) [reply src/_pytest/capture.py]: Done.

**Pull request merged**